components/webdriver_server/actions.rs

```diff
@@ -526,10 +514,13 @@ impl Handler {
         };
 
         // Step 3 - 4
-        // Get coordinates relative to an origin. Origin must be viewport.
+        // Get coordinates relative to an origin.
         let (x, y) = match action.origin {
             PointerOrigin::Viewport => (x_offset, y_offset),
-            _ => return Err(ErrorStatus::InvalidArgument),
+            PointerOrigin::Pointer => return Err(ErrorStatus::InvalidArgument),
+            PointerOrigin::Element(ref web_element) => {
+                self.get_element_origin_relative_coordinates(web_element)?
+            },
```

Comment:
```suggestion
            PointerOrigin::Element(ref web_element) => {
                self.get_element_origin_relative_coordinates(web_element)?
            },
            PointerOrigin::Pointer => return Err(ErrorStatus::InvalidArgument),
```
nit: let's don't put error handling in the middle of a match statement

Reply:
Hmm, the PR is merged. Should we open new PR or just wait until someone refactor?

Reply:
I think Xiaocheng means it is something we can pay attention to in the future :)

Reply:
Oh I see. Okay" :D